### PR TITLE
feat(badgeable): Pass down props to children

### PR DESCRIPTION
**Pull request by stevenzyang**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @stevenzyang has signed the CLA. Thanks for the pull request!

File: src/components/badgeable/__tests__/Badgeable-test.js [hunk 1]
```diff
@@ -10,6 +10,18 @@ describe('components/badgeable/Badgeable', () => {
         expect(wrapper).toMatchSnapshot();
     });
 
+    test('should pass down props to children when it is an element', () => {
+        const children = <div id="render">Some text to render</div>;
+        const wrapper = shallow(<Badgeable foo="bar">{children}</Badgeable>);
+        // console.log(wrapper.prop("foo"));
```
Comment (booc0mtaco) [inline src/components/badgeable/__tests__/Badgeable-test.js]: take this line out to tidy up

File: src/components/badgeable/__tests__/Badgeable-test.js [hunk 2]
```diff
@@ -10,6 +10,18 @@ describe('components/badgeable/Badgeable', () => {
         expect(wrapper).toMatchSnapshot();
     });
 
+    test('should pass down props to children when it is an element', () => {
+        const children = <div id="render">Some text to render</div>;
+        const wrapper = shallow(<Badgeable foo="bar">{children}</Badgeable>);
```
Comment (booc0mtaco) [inline src/components/badgeable/__tests__/Badgeable-test.js]: To make this more deliberate, let's `shallow` and include the output for `children` in the wrapper directly.

```
shallow(
  <Badgeable foo="bar">
    <div id="render">etc.</div>
  </Badgeable>
);

Let's also add a test to define what should happen if `children` is more than one Element.

File: src/components/badgeable/Badgeable.js [hunk 3]
```diff
@@ -18,11 +18,19 @@ type Props = {
...
+        topRight = null,
+        bottomLeft = null,
+        bottomRight = null,
+        ...rest
+    } = props;
 
     return (
         <div className={`badgeable-container ${className}`}>
-            {children}
+            {React.isValidElement(children) ? React.cloneElement(React.Children.only(children), rest) : children}
```
Comment (jstoffan) [inline src/components/badgeable/Badgeable.js]: What is the purpose of this change? I would think that since `children` should always be declarative (eslint rule), props could be passed directly rather than via `Badgeable`.

Comment (stevenzyang) [reply src/components/badgeable/Badgeable.js]: We have a pattern in EUA of using `HeaderToggleButton` to wrap a button and pass a `color` prop depending on the state of the header and the theme. For NotificationCenterButton and TaskFlyout we wrap these buttons in a `Badgeable` as well. Due to this, there's currently a bug where the the inner icon element is not receiving the correct color prop from `HeaderToggleButton`, and this would fix that issue.

File: src/components/badgeable/__tests__/Badgeable-test.js [hunk 4]
```diff
@@ -4,12 +4,37 @@ import Badgeable from '../Badgeable';
...
+        );
+        expect(
+            wrapper
+                .find('#render')
+                .first()
+                .prop('foo'),
+        ).toEqual('bar');
+    });
+
+    test('should throw an error when there are multiple children', () => {
```
Comment (jstoffan) [inline src/components/badgeable/__tests__/Badgeable-test.js]: The component is losing the ability to render text and multiple components, which seems like a breaking change? Does these features get used anywhere internally or externally?

File: src/components/badgeable/Badgeable.js (hunk 3 above)
Comment (jstoffan) [reply src/components/badgeable/Badgeable.js]: It sounds like the `HeaderToggleButton` component should be improved to allow arbitrary content to be rendered rather than decorated, if that's the case. Would it be possible to add a render prop to `HeaderToggleButton` so you can access the `color` prop?

**Comment by stevenzyang** [comment]
Following @jstoffan's suggestion to modify HeaderToggleButton to accept a render prop instead
